Makefile

```diff
@@ -1,51 +1,77 @@
-OS_NAME := $(shell uname -s | tr A-Z a-z)
-EXEC = bin/LRez
-ifeq ($(OS_NAME),linux)
-        SHLIB_EXT=so
-else
-        SHLIB_EXT=dylib
+ifeq ($(SHLIB_EXT),)
+	OS_NAME := $(shell uname -s | tr A-Z a-z)
+	ifeq ($(OS_NAME),linux)
+		SHLIB_EXT=.so
+	else
+		SHLIB_EXT=.dylib
+	endif
 endif
 
 curDir = $(shell pwd)
-CC = g++
-CFLAGS  = -Wall -pedantic -O3 -m64 -shared -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -std=c++11 -fPIC
 
-BAMTOOLS_INC = $(curDir)/bamtools/include/bamtools/
-BAMTOOLS_LIB = $(curDir)/bamtools/lib/
+PREFIX ?= /usr/local
+BINDIR = $(PREFIX)/bin
+LIBDIR = $(PREFIX)/lib
```

Comment:
These are some common paths so that a `make install` will use the standard `/usr/local` location, unless overridden.